expression/expressions/helper.go

```diff
@@ -498,22 +498,23 @@ func CheckAllOneColumns(args ...expression.Expression) error {
 	return nil
 }
 
-func hasSameColumn(lhs expression.Expression, rhs expression.Expression) error {
-	l, okl := lhs.(*Row)
-	r, okr := rhs.(*Row)
-	if okl && okr {
-		if len(l.Values) != len(r.Values) {
-			return errors.Errorf("Operand should contain %d column(s)", len(l.Values))
+func columnNumber(e expression.Expression) int {
+	v, ok := e.(*Row)
+	if ok {
+		// TODO: add check, row constructor must have >= 2 columns
+		return len(v.Values)
+	}
+
+	return 1
+}
+
+func hasSameColumn(e expression.Expression, args ...expression.Expression) error {
```

Comment:
seems hasSameColumn only check column number length, how about rename to `hasSameColumnLen`?